lib/dal/src/socket/input.rs

```diff
@@ -238,26 +234,29 @@ impl InputSocket {
             let node_weight =
                 NodeWeight::new_content(change_set, id, ContentAddress::InputSocket(hash))?;
             workspace_snapshot.add_node(node_weight).await?;
-            workspace_snapshot
-                .add_edge(
-                    schema_variant_id,
-                    EdgeWeight::new(change_set, EdgeWeightKind::Socket)?,
-                    id,
-                )
-                .await?;
+            SchemaVariant::add_edge_to_input_socket(
+                ctx,
+                schema_variant_id,
+                id.into(),
+                EdgeWeightKind::Socket,
+            )
+            .await
+            .map_err(Box::new)?;
         }
 
         let attribute_prototype = AttributePrototype::new(ctx, func_id).await?;
 
-        ctx.workspace_snapshot()?
-            .add_edge(
-                id,
-                EdgeWeight::new(change_set, EdgeWeightKind::Prototype(None))?,
-                attribute_prototype.id(),
-            )
-            .await?;
+        let socket = Self::assemble(id.into(), content);
+
+        Self::add_edge_to_attribute_prototype(
+            ctx,
+            socket.id,
+            attribute_prototype.id(),
+            EdgeWeightKind::Prototype(None),
+        )
+        .await?;
 
-        Ok(Self::assemble(id.into(), content))
+        Ok(socket)
```

Comment:
Why do some of these "create_edge + assemble and return"  blocks get reordered? Does that change functionality?

Reply:
Not really, it's mostly that we can use the typed socket.id instead of calling id.into(), it's also side-effect from the previous implementation that required `&self` to opperate. Since this changed almost 30 files I wasn't that thorough to ensure the diff was minimized.

But should not change functionality

Reply:
ahh gotcha! looks good